### PR TITLE
Notification system (Work in progress, not yet to be merged)

### PR DESCRIPTION
**Pull request by furilo**
I've created the basic model for subscriptions. Things that are already being done: 

- Issue owner gets subscribed when she creates the issue
- Users get subscribed when they add a reason or a vote
- Users have a new attribute (email_subscription_mode)
- Subscribers to an issue receive a new notification to their queue when something (new vote, new reason) happens in their subscribed issue. 

@ferblape (or @bertocq? :) I'd need some help: 

- Review that all the previous is working fine. Specially migrations might not be working properly as I got a little messed up over there.
- I'm blocked with one thing. To update the subscription I'm putting this link: 

```
<%= link_to "<span><strong>Following</strong> Hourly digest</span> Group them and receive no more than one mail per hour".html_safe, 
  update_subscription_path(@subscription, :email_subscription_mode => 'hourly'), 
  :remote => true, :method => 'patch' %>
```

But when you send it (interaction is not finished, but you can see in the log what gets sent) this parameters are received:  
```
Parameters: {"email_subscription_mode"=>"daily", "id"=>"6"}
```

And because of that it fails to get updated. I understand that the subscription object should be received as parameter (instead of only the id). I tried different things but I am failing miserably. What am I doing wrong? Tx!

### CODE REVIEW

**Comment by furilo** [comment]
(There are several `ToReview` comments in the code which, um, would welcome review ;)

File: app/models/subscription.rb [hunk 1]
```diff
@@ -0,0 +1,15 @@
+class Subscription < ActiveRecord::Base
+
+  belongs_to :issue
+  belongs_to :user
+
+  # ToReview: I can't get the system to save this properly, it always save 0 as the value
```
Comment (ferblape) [inline app/models/subscription.rb]: That's because you have an incorrect schema in the database. You might need to clean your database and seed it again.

**Comment by ferblape** [comment]
I'm reviewing this @bertocq, but I haven't committed anything yet.

**Comment by ferblape** [comment]
This is ready to be picked again. Some comments:

1 - the migrations were wrong, because the database type for enumerations is integer and not string. I have updated the migrations. In your laptop you need to run this command to redo the last 3 migrations:

```
bin/rake db:migrate:redo STEP=3
```

2 - I have extracted most of the calls to `subscribe` to a class method in the `Subscriptions` class. Repetition has been minimized

3 - to remove it completely we should open the Voting class and add a callback, maybe that's something we can do in the future

4 - in the subscriptions controller, you were calling to `permit` in a flat params structure. That only works when the params are nested

You can continue from here. Ping me if you need help!

**Comment by ferblape** [comment]
Notifications released. Let's test them!

**Comment by ferblape** [comment]
Implements #4 and #5

**Pull request merged**